FirebaseStorage/Tests/Unit/FIRStorageDeleteTests.m

```diff
@@ -102,7 +102,29 @@ - (void)testSuccessfulFetch {
   [FIRStorageTestHelpers waitForExpectation:self];
 }
 
-- (void)testUnsuccessfulFetchUnauthenticated {
+- (void)testSuccessfulFetchWithEmulator {
+  XCTestExpectation *expectation =
+      [self expectationWithDescription:@"testSuccessfulFetchWithEmulator"];
+
+  [self.storage useEmulatorWithHost:@"localhost" port:8080];
+  self.fetcherService.testBlock =
+      [FIRStorageTestHelpers successBlockWithURL:@"https://localhost:8080/v0/b/bucket/o/object"];
+
+  FIRStoragePath *path = [FIRStorageTestHelpers objectPath];
+  FIRStorageReference *ref = [[FIRStorageReference alloc] initWithStorage:self.storage path:path];
+  FIRStorageDeleteTask *task = [[FIRStorageDeleteTask alloc] initWithReference:ref
+                                                                fetcherService:self.fetcherService
+                                                                 dispatchQueue:self.dispatchQueue
+                                                                    completion:^(NSError *error) {
+                                                                      XCTAssertEqual(error, nil);
+                                                                      [expectation fulfill];
+                                                                    }];
+  [task enqueue];
+
+  [FIRStorageTestHelpers waitForExpectation:self];
+}
+
+- (void)testUnsuccessfulDeleteUnauthenticated {
   XCTestExpectation *expectation =
       [self expectationWithDescription:@"testUnsuccessfulFetchUnauthenticated"];
```

Comment:
nit: change the expectation name too

Reply:
I changed the test name back (all the test use Fetch). I wanted to clean this up originally but then gave up.